tufrepo/cli.py

```diff
@@ -115,6 +115,39 @@ def snapshot(ctx: Context):
     ctx.obj.repo.snapshot()
 
 
+@cli.command()
+@click.pass_context
+@click.argument("role")
+def init_succinct_roles(ctx: Context, role: str):
+    """Create a new set of delegated bin files based on the succinct hash bin
+    delegation information in ROLE.
+
+    Note: it's required to call 'tufrepo edit ROLE add_succinct_delegation'
+    command first."""
+
+    targets: Targets
+    with ctx.obj.repo.edit(role) as targets:
+        if targets.delegations is None or targets.delegations.succinct_roles is None:
+            raise ValueError(
+                "Call 'tufrepo edit ROLE add_succinct_delegation' command first"
+            )
+
+        full_keys_info: List[PrivateKey] = []
+        for _ in range(targets.delegations.succinct_roles.threshold):
+            new_key: PrivateKey = ctx.obj.keyring.generate_key()
+            full_keys_info.append(new_key)
+            helpers.add_key(targets, role, None, new_key.public)
+
+        for bin_name in targets.delegations.succinct_roles.get_roles():
+            # Store all generated keys for delegations for each of the bins.
+            for key in full_keys_info:
+                ctx.obj.keyring.store_key(bin_name, key)
```

Comment:
the way privkeys.json size explodes here leads me to think the format is not a great design... but the whole privkeys.json is a hack in any case: succinct delegations are meant for environment keys anyway

No need to do anything about this here.